replica/table.cc

```diff
@@ -2923,6 +2923,16 @@ write_memtable_to_sstable(memtable& mt, sstables::shared_sstable sst) {
     }
 }
 
+void table::hold_gate_for_compaction_groups(const dht::partition_range& pr, utils::small_vector<gate::holder, 2>& async_gate_holders) {
```

Comment:
minor: it would be more elegant if we returned a utils::small_vector<gate::holder, 1>, that could be a single entry vector in the singular range case, or otherwise, we can reserve the exact number of items based on the size of result of compaction_groups_for_token_range.

This way, the caller can keep the table's gate holder in a simple variable
and the rest either as a single small_vector in `table::mutation_query` or a vector of vectors in `table::query`
where the top-level vector size is also known in advance as partition_ranges.size()

Reply:
Done in v3.